pkg/api/validation/validation_test.go

```diff
@@ -3792,7 +3923,52 @@ func TestValidateMountPropagation(t *testing.T) {
 			t.Errorf("test %d expected success, got error: %v", i, errs)
 		}
 	}
+}
+
+func TestAlphaValidateVolumeDevices(t *testing.T) {
+	volumes := sets.NewString("abc", "abc-123", "def", "ghi", "jkl")
+
+	disabledAlphaVolDevice := []api.VolumeDevice{
+		{Name: "abc", DevicePath: "/foo"},
+	}
+
+	successCase := []api.VolumeDevice{
+		{Name: "abc", DevicePath: "/foo"},
+		{Name: "abc-123", DevicePath: "/usr/share/test"},
+	}
+
+	errorCases := map[string][]api.VolumeDevice{
+		"empty name":           {{Name: "", DevicePath: "/foo"}},
```

Comment:
What about duplicate name?

Reply:
ah, good catch

Reply:
ok, think I addressed both of those concerns